flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/LocalStandaloneFlinkResourceFactory.java

```diff
@@ -29,9 +33,21 @@
 public final class LocalStandaloneFlinkResourceFactory implements FlinkResourceFactory {
 	private static final Logger LOG = LoggerFactory.getLogger(LocalStandaloneFlinkResourceFactory.class);
 
+	private static final ParameterProperty<Path> DISTRIBUTION_DIRECTORY = new ParameterProperty<>("distDir", Paths::get);
+	private static final ParameterProperty<Path> DISTRIBUTION_LOG_BACKUP_DIRECTORY = new ParameterProperty<>("logBackupDir", Paths::get);
+
 	@Override
 	public Optional<FlinkResource> create(FlinkResourceSetup setup) {
+		Optional<Path> distributionDirectory = DISTRIBUTION_DIRECTORY.get();
+		if (!distributionDirectory.isPresent()) {
+			LOG.warn("The distDir property was not set. You can set it when running maven via -DdistDir=<path> .");
```

Comment:
Does it mean that running e2e tests from IDE will always need to set `distDir` manually?

Reply:
yes, this isn't new.